Content.Server/Chat/Systems/ChatSystem.Emote.cs

```diff
@@ -88,6 +91,18 @@ public void TryEmoteWithChat(
         if (!forceEmote && !AllowedToUseEmote(source, emote))
             return;
 
+        // Check if this emote should be blocked. Forced emotes ignore this check.
+        if (!forceEmote)
+        {
+            var ev = new GetEmoteBlockersEvent();
+            RaiseLocalEvent(source, ref ev);
+            if (ev.ShouldBlock(emote))
```

Comment:
I haven't seen this way of doing this and I'm not sure its the correct approach! This really limits what the event can do (only blocking) and is confusing.

My recommendation:
Rename the event to something like `ModifyEmoteEvent`.
Send the emote itself, the source, and a field for if the popup should be displayed or not, in the event itself.
If the event is canceled, cancel the event and display the popup if the field is true.

Otherwise, just use whatever emote that is stored inside the event! If a system modified it, then you will use the new one.

This could allow for stuff like medical masks to make the cough sound be different if the mask is on or if you are wearing gloves, the snap sound is muffled etc...

What you do here also forces you to store the hash sets inside the event which is odd.

This also was never triggered in testing when I screamed in the chat! Not sure this is the correct location to put it.

Reply:
> This also was never triggered in testing when I screamed in the chat! Not sure this is the correct location to put it.

Perhaps I'm misunderstanding, but that's the difference between the Scream _action_ and the scream _emote_, isn't it?
As far as I know, the action is only triggered by hitting the hotkey on the top right.

Reply:
I honestly don't know the specifics, but I thought this pr would be for blocking all noises (action or emotes). Isn't the only action emote the scream?